src/Microsoft.Health.Fhir.Shared.Api/Controllers/SearchParameterController.cs

```diff
@@ -0,0 +1,108 @@
+{
+    /// <summary>
+    /// Search Parameter State Controller
+    /// </summary>
+    [ServiceFilter(typeof(OperationOutcomeExceptionFilterAttribute))]
+    public class SearchParameterController : Controller
+    {
+        private readonly IMediator _mediator;
+        private readonly CoreFeatureConfiguration _coreFeaturesConfig;
+
+        public SearchParameterController(IMediator mediator, IOptions<CoreFeatureConfiguration> coreFeatures)
+        {
+            EnsureArg.IsNotNull(mediator, nameof(mediator));
+            EnsureArg.IsNotNull(coreFeatures?.Value, nameof(coreFeatures));
+
+            _mediator = mediator;
+            _coreFeaturesConfig = coreFeatures.Value;
+        }
+
+        /// <summary>
+        /// Gets all search parameters with current status.
+        /// </summary>
+        /// <param name="cancellationToken">Cancellation Token</param>
+        /// <returns>Parameters resource containing search parameters matching search query. Returns all search parameters with current state if no search query provided. Along with Status for each result.</returns>
+        [HttpGet]
+        [Route(KnownRoutes.SearchParametersStatusQuery, Name = RouteNames.SearchParameterState)]
+        [AuditEventType(AuditEventSubType.SearchParameterStatus)]
+        [ValidateSearchParameterStateRequestAtrribute]
+        public async Task<IActionResult> GetSearchParametersStatus(CancellationToken cancellationToken)
+        {
+            CheckIfSearchParameterStatusIsEnabledAndRespond();
+
+            SearchParameterStateRequest request = new SearchParameterStateRequest(GetQueriesForSearch());
+            SearchParameterStateResponse result = await _mediator.Send(request, cancellationToken);
+
+            _ = result ?? throw new ResourceNotFoundException(Resources.SearchParameterStatusNotFound);
+
+            return FhirResult.Create(result.SearchParameters, System.Net.HttpStatusCode.OK);
+        }
+
+        /// <summary>
+        /// Search for multiple Search Parameter Status using POST.
+        /// </summary>
+        /// <param name="cancellationToken">Cancellation Token</param>
+        /// <returns>Parameters resource containing Search Parameters with Status.</returns>
+        [HttpPost]
+        [Route(KnownRoutes.SearchParameters + KnownRoutes.SearchParametersStatusPostQuery, Name = RouteNames.PostSearchParameterState)]
+        [ValidateParametersResource]
+        [AuditEventType(AuditEventSubType.SearchParameterStatus)]
+        public async Task<IActionResult> PostSearchParametersStatus(CancellationToken cancellationToken)
+        {
+            CheckIfSearchParameterStatusIsEnabledAndRespond();
+
+            SearchParameterStateRequest request = new SearchParameterStateRequest(GetQueriesForSearch());
+            SearchParameterStateResponse result = await _mediator.Send(request, cancellationToken);
+
+            _ = result ?? throw new ResourceNotFoundException(Resources.SearchParameterStatusNotFound);
```

Comment:
Fancy `if` statement? :)

Reply:
Yes, this is just a check since I intentionally send back as a null to trigger the not found. If there is a better way I can update that.